prescient/model_components/data_loader.py

```diff
@@ -242,10 +242,9 @@ def nd_gen_init(m,b):
     model.NondispatchableGeneratorsAtBus = Set(model.Buses, initialize=nd_gen_init)
     
     def NonNoBus_init(m):
-        retval = set()
         for b in m.Buses:
-            retval = retval.union([gen for gen in m.NondispatchableGeneratorsAtBus[b]])
-        return retval
+            for g in m.NondispatchableGeneratorsAtBus[b]:
+                yield g
     
```

Comment:
Is each generator in exactly one bus?  If so this is good, if not the old code prevented duplicates.

Reply:
Yes, it's a data error for it to be otherwise. The Pyomo `Set` object will now at least print a warning if their are duplicates (and discard them), whereas before they were silently ignored.